src/layout/tile.rs

```diff
@@ -967,14 +968,44 @@ impl<W: LayoutElement> Tile<W> {
                     if clip_to_geometry {
                         if let Some(shader) = clip_shader.clone() {
                             if ClippedSurfaceRenderElement::will_clip(&elem, scale, geo, radius) {
-                                return ClippedSurfaceRenderElement::new(
-                                    elem,
-                                    scale,
-                                    geo,
-                                    shader.clone(),
-                                    radius,
-                                )
-                                .into();
+                                match elem.texture() {
+                                    WaylandSurfaceTexture::Texture(_) => {
+                                        return ClippedSurfaceRenderElement::new(
+                                            elem,
+                                            scale,
+                                            geo,
+                                            shader.clone(),
+                                            radius,
+                                        )
+                                        .into();
+                                    }
+                                    WaylandSurfaceTexture::SolidColor(color32_f) => {
+                                        // In this branch we're rendering a solid color texture
+                                        // which smithay will render using a solid program shader.
+                                        // `ClippedSurfaceElement` is implemented by overriding the
+                                        // default texture program shader that smithay uses. This
+                                        // strategy won't work for solid color textures because
+                                        // smithay doesn't provide a way to override the solid
+                                        // program shader. To work around this, we can use a
+                                        // `BorderRenderElement` when we detect that we need to clip
+                                        // a solid color texture.
+                                        return BorderRenderElement::new(
+                                            geo.size,
+                                            Rectangle::from_size(geo.size),
+                                            GradientInterpolation::default(),
+                                            Color::from_color32f(*color32_f),
+                                            Color::from_color32f(*color32_f),
```

Comment:
Is this premultiplied? Does BorderRenderElement::new expect premultiplied? Add a comment here pls

Reply:
`BorderRenderElement` calls `Color::to_array_unpremul` on these, so it expects premultiplied, also verified that the border fragment shader expects unpremultiplied, then internally premultiplies them

I checked that the `color32_f` matches the single pixel buffer data which would mean its also premultiplied